src/Sdk/StrixMusic.Sdk/FileMetadata/FileMetadataManager.cs

```diff
@@ -27,13 +28,15 @@ public sealed class FileMetadataManager : IFileMetadataManager
     {
         private static string NewGuid() => Guid.NewGuid().ToString();
 
+        private readonly List<IFileData> _knownFiles = new();
         private readonly IFileScanner _fileScanner;
         private readonly AudioMetadataScanner _audioMetadataScanner;
         private readonly PlaylistMetadataScanner _playlistMetadataScanner;
         private readonly IFolderData _rootFolderToScan;
         private readonly IFolderData _metadataStorage;
         private readonly IProgress<FileScanState>? _scanProgress;
         private CancellationTokenSource? _inProgressScanCancellationTokenSource;
+        private Task? _fileScannerTask;
```

Comment:
Still need to assign this when starting/stopping a scan.

Could potentially optimize it so that if there's an ongoing scan, it waits for the file to be found or the scan to finish (whichever comes first)